tests/integration-tests/tests/storage/test_efs.py

```diff
@@ -68,6 +73,145 @@ def test_efs_same_az(region, scheduler, pcluster_config_reader, clusters_factory
     _test_efs_correctly_shared(remote_command_executor, mount_dir, scheduler_commands)
 
 
+@pytest.mark.dimensions("ap-northeast-2", "c5.xlarge", "alinux", "awsbatch")
+@pytest.mark.dimensions("ap-northeast-2", "c5.xlarge", "alinux2", "slurm")
+@pytest.mark.usefixtures("os", "instance")
+def test_existing_efs(
+    region,
+    scheduler,
+    existing_efs,
+    pcluster_config_reader,
+    clusters_factory,
+    vpc_stack,
+    request,
+    key_name,
+    cfn_stacks_factory,
+):
+    """
+    Test when efs_fs_id is provided in the config file, the existing efs can be correctly mounted.
+
+    To verify the efs is the existing efs, the test expects a file with random ran inside the efs mounted
+    """
+    random_file_name = _write_file_into_efs(region, vpc_stack, existing_efs, request, key_name, cfn_stacks_factory)
+
+    _assert_subnet_az_relations(region, vpc_stack, expected_in_same_az=False)
+    mount_dir = "/efs_mount_dir"
+    cluster_config = pcluster_config_reader(
+        mount_dir=mount_dir, efs_fs_id=existing_efs.cfn_resources["FileSystemResource"]
+    )
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+
+    # test file in efs exist
+    logging.info("Testing efs {0} is correctly mounted".format(mount_dir))
+    result = remote_command_executor.run_remote_command("df | grep '{0}'".format(mount_dir))
+    assert_that(result.stdout).contains(mount_dir)
+
+    remote_command_executor.run_remote_command(f"cat {mount_dir}/{random_file_name}")
+    scheduler_commands = get_scheduler_commands(scheduler, remote_command_executor)
+    _test_efs_correctly_mounted(remote_command_executor, mount_dir)
+    _test_efs_correctly_shared(remote_command_executor, mount_dir, scheduler_commands)
+    remote_command_executor.run_remote_command(f"cat {mount_dir}/{random_file_name}")
+
+
+@pytest.fixture(scope="class")
+def existing_efs(cfn_stacks_factory, request, region):
+    # efs stack contains a single efs resource
```

Comment:
minor: let's use python triple quotes syntax for method comment.

Reply:
Done